### ISSUE TITLE
7.26.0 Upgrade Issues - CrashBackLoop

### ISSUE THREAD

**Issue by rimbesat** [DataDog/chaos-controller#747]
**Describe the bug**
Hi team, We recently tried to upgrade from version 7.13.1 to 7.26 and we are facing issues trying to achieve the same behaviour. We previously reached out to you about [issues with dynamic targeting](https://github.com/DataDog/chaos-controller/issues/744), we upgraded to your newest release and we are hoping to get some further guidance.
We found that running a container failure (graceful and forceful, dynamic and static targeting) causes the targeted pods to go into a CrashBackLoop for the duration of the experiment

**To Reproduce**
1. This is an example of a forceful container failure experiment
```
apiVersion: chaos.datadoghq.com/v1beta1
kind: Disruption
metadata:
  name: container-failure-all-forced
  # Namespace should be your application namespace.
  namespace: chaos-engineering-framework
  annotations:
    chaos.datadoghq.com/environment: "dev"
spec:
  level: pod
  duration: 6m0s
  selector:
    app.kubernetes.io/name: "fault-injection-showcase"
    app: "fault-injection-showcase--plugin-master"
  count: 100%
  #setting staticTargeting as true is optional and is required only if you want to execute experiments with static targeting
  staticTargeting: true
  containerFailure:
    forced: true
```

For the container forced experiment we found
(Injector logs)
```
{"level":"warn","ts":1690464275041.3286,"caller":"injector/main.go:620","message":"couldn't watch targeted pod, retrying error an error occurred during reinjection: unable to reinitialize netns manager and cgroup manager for containerID 0809c40b517ade8875f9b3b55b9793474914b462e4a1ebe82c1c9832bd18c296 (PID: 15431): error creating network namespace manager: error getting given PID (15431) network namespace from path /mnt/host/proc/15431/ns/net: no such file or directory retrying 589.973741ms","disruptionName":"container-failure-all-forced","disruptionNamespace":"chaos-engineering-framework","targetName":"fault-injection-showcase-plugin-master-template-b785fd95b-lzbzh","targetNodeName":"ip-10-72-65-218.us-west-2.compute.internal"}
```
This could be related to our shift to using containerd over dockershim, due to our kubernetes upgrade to 1.24
**Expected behavior**
We expect the targeted pods not to go into a CrashBackLoop for container failure.
Let us know if our expectations are incorrect.

**Logs**
[chaos-controller-27-06.txt](https://github.com/DataDog/chaos-controller/files/12185863/chaos-controller-27-06.txt)

[injector_container-failure-forced.txt](https://github.com/DataDog/chaos-controller/files/12185868/injector_container-failure-forced.txt)

**Environment:**
Kubernetes version: v1.24.15-eks-a5565ad
Controller version: 7.26.0
Cloud provider (or local): AWS EKS
Base OS for Kubernetes: Amazon Linux: 5.4.242-156.349.amzn2.x86_64

**Comment by ptnapoleon** [comment]
I think this might be expected in some situations, let me investigate. If we're continually causing the container failure every time the target starts up, that would definitely lead it to a CLB state

**Comment by ptnapoleon** [comment]
Okay, I've confirmed that this is the expected situation from our point of view. We will continually cause container failures on the targeted pod for the duration of the disruption. Given you're using staticTargeting, if the target pods were to be rescheduled in any way, we would not target the new pods, but as long as the original targets remain, and it's just the internal containers being restarted, we will keep identifying the new container and killing it.

I went back to 7.13.1 to investigate the change in behavior, and best as I can tell, it seems we have always _intended_ for this to be the behavior, but there race conditions where if a container restarted too quickly, we could lose track of the new container id, and would choose to stop injecting altogether. That race condition has been resolved, and now the container failure should be much more durable.

This is the preferred behavior from most users, but if you'd like this issue to become a feature request for a way to configure container failures to be "at most once", just let me know. Thanks!
